src/output/restart_file.jl

```diff
@@ -1,44 +1,53 @@
+export RestartFile
+
+"""RestartFile callback. Writes a restart file as .jld2.
+Options are $(TYPEDFIELDS)"""
+@kwdef mutable struct RestartFile <: AbstractCallback
+    "[OPTION] File name for restart file, should end with .jld2"
+    filename::String = "restart.jld2"
+
+    "[OPTION] Path for restart file, uses model.output.run_path if not specified"
+    path::String = ""
+
+    "[OPTION] Apply lossless compression in JLD2?"
+    compress::Bool = true
+
+    "[OPTION] Only write with model.output.active = true?"
+    write_only_with_output::Bool = true
+
+    "[DERIVED] package version to track possible incompatibilities"
+    pkg_version::VersionNumber = isnothing(pkgversion(SpeedyWeather)) ? v"0.0.0" : pkgversion(SpeedyWeather)
+end
+
+initialize!(::RestartFile, args...) = nothing
+callback!(::RestartFile, args...) = nothing
+
 """
 $(TYPEDSIGNATURES)
 A restart file `restart.jld2` with the prognostic variables is written
 to the output folder (or current path) that can be used to restart the model.
-`restart.jld2` will then be used as initial conditions. The prognostic variables
-are bitrounded for compression and the 2nd leapfrog time step is discarded.
-Variables in restart file are unscaled."""
-function write_restart_file!(
-    output::AbstractOutput,
+`restart.jld2` will then be used as initial conditions.
+Variables in restart file are not scaled."""
+function finalize!(
+    restart::RestartFile,
     progn::PrognosticVariables,
+    diagn::DiagnosticVariables,
+    model::AbstractModel,
 )
-    
-    # exit immediately if no output or no restart file desired
-    output.active && output.write_restart || return nothing  
-    
-    # move 2nd leapfrog to 1st to compress restart file
-    get_step(progn.vor, 1)   .= get_step(progn.vor, 2)
-    get_step(progn.div, 1)   .= get_step(progn.div, 2)
-    get_step(progn.temp, 1)  .= get_step(progn.temp, 2)
-    get_step(progn.humid, 1) .= get_step(progn.humid, 2)
-    get_step(progn.pres, 1)  .= get_step(progn.pres, 2)
-
-    # bitround 1st leapfrog step to output precision
-    if eltype(progn) <: Base.IEEEFloat  # currently not defined for other formats...
```

Comment:
why was that here, and now it's not needed anymore?

Reply:
There was some lossy compression for `restart.jld2` which I thought was just consistent with the NetCDFOutput compression but actually it makes very little difference in terms of overall output data size and interferes with being able to restart with leapfrog so I just took it out and there's no lossy compression anymore for the restart file. Only lossless but I don't it does much